app/views/shared/_list_posts.html.slim

```diff
@@ -1,17 +1,20 @@
 - posts.each do |post|
-  .p-2.mb-2.bg-white.border.rounded.shadow-sm
+  .mb-2.bg-white.border.rounded.shadow-sm
     - if post.pending?
-      .px-4.text-warning
+      .px-4.pt-2.text-warning
         | Pending 
     - if post.rejected?
-      .px-4.text-danger
+      .px-4.pt-2.text-danger
         | Rejected
+    - if page.present? && page == 1 && post == posts.first && post.cover_image.attached?
```

Comment:
You can create a helper method instead of putting complicated logic into view partial

https://www.rubyguides.com/2020/01/rails-helpers